appservice/src/deploy/runWithZipStream.ts

```diff
@@ -55,6 +55,12 @@ export async function runWithZipStream(context: IActionContext, fsPath: string,
             zipFile.addFile(fsPath, path.basename(fsPath));
         }
 
+        zipFile.outputStream.on('finish', () => {
```

Comment:
It looks like you could use `on.('data', ...` instead to manually add up all the chunks like shown [here](https://stackoverflow.com/a/48868017/7044639) for a similar package. I think that would be more comfy than relying on `outputStreamCursor` to be there

Reply:
This ends up being slightly inaccurate due to measuring the buffer size.  While close, it's not the exact same thing as the content size.  Still investigating how measure that accurately.

Reply:
Okay, I don't think it's worth pursuing this any further.  I was able to find that when compressed, the `zipFile` object has `entries: Entry[]`.  This object type is not exposed by the public API (intentionally), but has a property called `compressedSize`.  Adding all the compressedSizes, the zipFile size difference between the two methods is like 0.02 MB.  (8.19 MB vs. 8.17 MB)

I'm not sure how Kudu is calculating their size since it reports about ~7.9 MB (it'll tell you it's deleting the old zipfile when you deploy).  But I think I've wasted enough time trying to get the numbers to match especially since they're close enough and it's probably all an estimate anyway.